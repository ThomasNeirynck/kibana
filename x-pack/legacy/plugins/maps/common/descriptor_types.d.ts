/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export type EMSXYZSourceType = 'EMS_XYZ';
export type SourceType = EMSXYZSourceType | string;

export interface ISourceDescriptor {
  id: string;
  type: SourceType;
}

export interface IXYZTMSSourceDescriptor extends ISourceDescriptor {
  urlTemplate: string;
  type: EMSXYZSourceType;
}

export interface ILayerDescriptor {
  sourceDescriptor: ISourceDescriptor;
  id: string;
}
