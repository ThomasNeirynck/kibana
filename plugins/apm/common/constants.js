export const APP_NAME = 'context.app.name';
export const APP_AGENT_NAME = 'context.app.agent.name';
export const EVENT_PROCESSOR_NAME = 'processor.name';

export const TRANSACTION_DURATION = 'transaction.duration.us';
export const TRANSACTION_TYPE = 'transaction.type';
export const TRANSACTION_RESULT = 'transaction.result';
export const TRANSACTION_NAME = 'transaction.name';
export const TRANSACTION_ID = 'transaction.id';

export const TRACE_START = 'trace.start.us';
export const TRACE_DURATION = 'trace.duration.us';
export const TRACE_TRANSACTION_ID = 'trace.transaction_id';
export const TRACE_TYPE = 'trace.type';
export const TRACE_NAME = 'trace.name';
export const TRACE_ID = 'trace.id';
export const TRACE_SQL = 'context.db.statement';

export const ERROR_GROUP_ID = 'error.grouping_key';
export const ERROR_CULPRIT = 'error.culprit';
export const ERROR_MESSAGE = 'error.exception.message';
