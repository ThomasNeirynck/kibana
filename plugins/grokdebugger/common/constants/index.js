export { ROUTES } from './routes';
export { PLUGIN } from './plugin';
