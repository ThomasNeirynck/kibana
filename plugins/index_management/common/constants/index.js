export { PLUGIN } from './plugin';
export { BASE_PATH } from './base_path';
export * from './index_statuses';
