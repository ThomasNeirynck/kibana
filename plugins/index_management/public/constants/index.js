export { REFRESH_RATE_INDEX_LIST } from './refresh_intervals';
