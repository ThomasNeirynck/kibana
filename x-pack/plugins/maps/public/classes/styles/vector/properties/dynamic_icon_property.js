/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import _ from 'lodash';
import React from 'react';
import { assignCategoriesToPalette } from '../style_util';
import { DynamicStyleProperty } from './dynamic_style_property';
import { getIconPalette, getMakiIconId, getMakiSymbolAnchor } from '../symbol_utils';
import { BreakedLegend } from './components/breaked_legend';

export class DynamicIconProperty extends DynamicStyleProperty {
  isOrdinal() {
    return false;
  }

  isCategorical() {
    return true;
  }

  getNumberOfCategories() {
    const palette = getIconPalette(this._options.iconPaletteId);
    return palette ? palette.length : 0;
  }

  syncIconWithMb(symbolLayerId, mbMap, iconPixelSize) {
    if (this._isIconDynamicConfigComplete()) {
      mbMap.setLayoutProperty(
        symbolLayerId,
        'icon-image',
        this._getMbIconImageExpression(iconPixelSize)
      );
      mbMap.setLayoutProperty(symbolLayerId, 'icon-anchor', this._getMbIconAnchorExpression());
    } else {
      mbMap.setLayoutProperty(symbolLayerId, 'icon-image', null);
      mbMap.setLayoutProperty(symbolLayerId, 'icon-anchor', null);
    }
  }

  _getPaletteStops() {
    if (this._options.useCustomIconMap && this._options.customIconStops) {
      const stops = [];
      for (let i = 1; i < this._options.customIconStops.length; i++) {
        const { stop, icon } = this._options.customIconStops[i];
        stops.push({
          stop,
          style: icon,
        });
      }

      return {
        fallbackSymbolId:
          this._options.customIconStops.length > 0 ? this._options.customIconStops[0].icon : null,
        stops,
      };
    }

    return assignCategoriesToPalette({
      categories: _.get(this.getCategoryFieldMeta(), 'categories', []),
      paletteValues: getIconPalette(this._options.iconPaletteId),
    });
  }

  _getMbIconImageExpression(iconPixelSize) {
    const { stops, fallback } = this._getPaletteStops();

    if (stops.length < 1 || !fallback) {
      //occurs when no data
      return null;
    }

    const mbStops = [];
    stops.forEach(({ stop, style }) => {
      mbStops.push(`${stop}`);
      mbStops.push(getMakiIconId(style, iconPixelSize));
    });
    mbStops.push(getMakiIconId(fallback, iconPixelSize)); //last item is fallback style for anything that does not match provided stops
    return ['match', ['to-string', ['get', this._field.getName()]], ...mbStops];
  }

  _getMbIconAnchorExpression() {
    const { stops, fallback } = this._getPaletteStops();

    if (stops.length < 1 || !fallback) {
      //occurs when no data
      return null;
    }

    const mbStops = [];
    stops.forEach(({ stop, style }) => {
      mbStops.push(`${stop}`);
      mbStops.push(getMakiSymbolAnchor(style));
    });
    mbStops.push(getMakiSymbolAnchor(fallback)); //last item is fallback style for anything that does not match provided stops
    return ['match', ['to-string', ['get', this._field.getName()]], ...mbStops];
  }

  _isIconDynamicConfigComplete() {
    return this._field && this._field.isValid();
  }

  renderLegendDetailRow({ isPointsOnly, isLinesOnly }) {
    const { stops, fallbackSymbolId } = this._getPaletteStops();

    return (
      <BreakedLegend
        style={this}
        isPointsOnly={isPointsOnly}
        isLinesOnly={isLinesOnly}
        stops={stops}
        symbolId={null}
        color={'grey'}
        useFallback={!!fallbackSymbolId}
        fallbackSymbolId={fallbackSymbolId}
        fallbackColor={null}
      />
    );
  }
}
