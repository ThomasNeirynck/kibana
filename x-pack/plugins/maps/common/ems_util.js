/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */


import { GIS_API_PATH } from './constants';

export  async function getEMSResources(emsClient, includeElasticMapsService, licenseUid, useCors) {

  if (!includeElasticMapsService) {
    return {
      fileLayers: [],
      tmsServices: []
    };
  }

  emsClient.addQueryParams({ license: licenseUid });
  const fileLayerObjs = await emsClient.getFileLayers();
  const tmsServicesObjs = await emsClient.getTMSServices();

  const fileLayers = fileLayerObjs.map(fileLayer => {
    //backfill to static settings
    const format = fileLayer.getDefaultFormatType();
    const meta = fileLayer.getDefaultFormatMeta();

    return {
      name: fileLayer.getDisplayName(),
      origin: fileLayer.getOrigin(),
      id: fileLayer.getId(),
      created_at: fileLayer.getCreatedAt(),
      attribution: fileLayer.getHTMLAttribution(),
      attributions: fileLayer.getAttributions(),
      fields: fileLayer.getFieldsInLanguage(),
      url: useCors ? fileLayer.getDefaultFormatUrl() : `../${GIS_API_PATH}/data/ems?id=${encodeURIComponent(fileLayer.getId())}`,
      format: format, //legacy: format and meta are split up
      meta: meta, //legacy, format and meta are split up,
      emsLink: fileLayer.getEMSHotLink()
    };
  });

  const tmsServices = tmsServicesObjs.map(tmsService => {
    return {
      origin: tmsService.getOrigin(),
      id: tmsService.getId(),
      minZoom: tmsService.getMinZoom(),
      maxZoom: tmsService.getMaxZoom(),
      attribution: tmsService.getHTMLAttribution(),
      attributionMarkdown: tmsService.getMarkdownAttribution(),
      url: tmsService.getUrlTemplate()
    };
  });

  return { fileLayers, tmsServices };
}
