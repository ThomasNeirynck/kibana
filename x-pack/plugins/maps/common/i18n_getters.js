/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { i18n } from '@kbn/i18n';
import { APP_TITLE_EN } from './constants';


export function getAppTitle() {
  return i18n.translate('xpack.maps.appTitle', {
    defaultMessage: APP_TITLE_EN
  });
}
