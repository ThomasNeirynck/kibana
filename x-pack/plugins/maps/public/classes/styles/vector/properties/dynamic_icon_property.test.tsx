/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

// eslint-disable-next-line max-classes-per-file
import { shallow } from 'enzyme';

jest.mock('ui/new_platform');
jest.mock('../components/vector_style_editor', () => ({
  VectorStyleEditor: () => {
    return <div>mockVectorStyleEditor</div>;
  },
}));

import React from 'react';
import { VECTOR_STYLES } from '../../../../../common/constants';
// @ts-ignore
import { DynamicIconProperty } from './dynamic_icon_property';
import { mockField, MockLayer } from './test_util';

const makeProperty = (options, field = mockField) => {
  return new DynamicIconProperty(options, VECTOR_STYLES.ICON, field, new MockLayer(), () => {
    return x => x + '_format';
  });
};

describe('DynamicIconProperty', () => {
  it('should derive category number from palettes', async () => {
    const filled = makeProperty({
      iconPaletteId: 'filledShapes',
    });
    expect(filled.getNumberOfCategories()).toEqual(6);
    const hollow = makeProperty({
      iconPaletteId: 'hollowShapes',
    });
    expect(hollow.getNumberOfCategories()).toEqual(5);
  });
});

test('Should render categorical legend with breaks', async () => {
  const iconStyle = makeProperty({
    iconPaletteId: 'filledShapes',
  });

  const legendRow = iconStyle.renderLegendDetailRow({ isPointsOnly: true, isLinesOnly: false });
  const component = shallow(legendRow);
  await new Promise(resolve => process.nextTick(resolve));
  component.update();

  expect(component).toMatchSnapshot();
});
