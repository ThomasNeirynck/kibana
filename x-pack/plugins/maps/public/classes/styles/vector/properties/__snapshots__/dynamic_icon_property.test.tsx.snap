// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render categorical legend with breaks 1`] = `
<div>
  <EuiFlexGroup
    gutterSize="xs"
    justifyContent="spaceBetween"
  >
    <EuiFlexItem
      grow={false}
    >
      <EuiToolTip
        content="foobar_label"
        delay="regular"
        position="top"
        title="Icon"
      >
        <EuiText
          className="eui-textTruncate"
          size="xs"
          style={
            Object {
              "maxWidth": "180px",
            }
          }
        >
          <small>
            <strong>
              foobar_label
            </strong>
          </small>
        </EuiText>
      </EuiToolTip>
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup
    direction="column"
    gutterSize="none"
  >
    <Category
      color="grey"
      isLinesOnly={false}
      isPointsOnly={true}
      key="0"
      label="US_format"
      styleName="icon"
      symbolId="circle"
    />
    <Category
      color="grey"
      isLinesOnly={false}
      isPointsOnly={true}
      key="1"
      label="CN_format"
      styleName="icon"
      symbolId="marker"
    />
    <Category
      color="grey"
      isLinesOnly={false}
      isPointsOnly={true}
      key="fallbackCategory"
      label={
        <EuiTextColor
          color="secondary"
        >
          Other
        </EuiTextColor>
      }
      styleName="icon"
      symbolId="square"
    />
  </EuiFlexGroup>
</div>
`;
